unifi/cams/reolink_nvr.py

```diff
@@ -0,0 +1,72 @@
+import tempfile
+
+import aiohttp
+import json
+
+from yarl import URL
+
+from unifi.cams.base import UnifiCamBase
+
+
+class ReolinkNVRCam(UnifiCamBase):
+    @classmethod
+    def add_parser(self, parser):
+        super(ReolinkNVRCam, self).add_parser(parser)
+        parser.add_argument("--username", "-u", required=True, help="NVR username")
+        parser.add_argument("--password", "-p", required=True, help="NVR password")
+        parser.add_argument("--channel", "-c", required=True, help="NVR camera channel")
+
+    def __init__(self, args, logger=None):
+        super().__init__(args, logger)
+        self.snapshot_dir = tempfile.mkdtemp()
+        self.motion_in_progress = False
+
+    async def get_snapshot(self):
+        img_file = "{}/screen.jpg".format(self.snapshot_dir)
+        url = f"http://{self.args.ip}/api.cgi?cmd=Snap&user={self.args.username}&password={self.args.password}&rs=6PHVjvf0UntSLbyT&channel={self.args.channel}"
+        try:
+            async with aiohttp.request("GET", url) as resp:
+                with open(img_file, "wb") as f:
+                    while True:
+                        chunk = await resp.content.read(1024)
+                        if not chunk:
+                            break
+                        f.write(chunk)
+        except aiohttp.ClientError:
+            self.logger.error("Failed to get snapshot")
```

Comment:
If you rebase, there's now a helper function since this seemed pretty common:

```suggestion
        await self.fetch_to_file(url, img_file)
```